src/components/structures/auth/ForgotPassword.tsx

```diff
@@ -136,7 +134,14 @@ export default class ForgotPassword extends React.Component<IProps, IState> {
                 serverIsAlive: true,
             });
         } catch (e) {
-            this.setState(AutoDiscoveryUtils.authComponentStateForError(e, "forgot_password") as IState);
+            const {
+                serverIsAlive,
+                serverDeadError,
+            } = AutoDiscoveryUtils.authComponentStateForError(e, "forgot_password");
```

Comment:
the TS strict check on this line might want `e` to be defined as `: any`, I guess.